presto-tests/src/main/java/com/facebook/presto/tests/statistics/StatisticsAssertion.java

```diff
@@ -0,0 +1,99 @@
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.plan.OutputNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import org.intellij.lang.annotations.Language;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.MATCH;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_BASELINE;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_ESTIMATE;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.defaultTolerance;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertTrue;
+
+public class StatisticsAssertion
+{
+    private final DistributedQueryRunner runner;
+
+    public StatisticsAssertion(DistributedQueryRunner runner)
+    {
+        this.runner = requireNonNull(runner, "runner is null");
+    }
+
+    public Result check(@Language("SQL") String query)
+    {
+        return new Result(metricComparisons(query));
+    }
+
+    public List<MetricComparison> metricComparisons(@Language("SQL") String query)
+    {
+        String queryId = runner.executeWithQueryId(runner.getDefaultSession(), query).getQueryId();
+        Plan queryPlan = runner.getQueryPlan(new QueryId(queryId));
+        StageInfo stageInfo = runner.getQueryInfo(new QueryId(queryId)).getOutputStage().get();
+        return new MetricComparator().getMetricComparisons(queryPlan, stageInfo);
+    }
+
+    public static class Result
+    {
+        private static final Predicate<PlanNode> IS_OUTPUT_NODE = OutputNode.class::isInstance;
+
+        private final List<MetricComparison> metricComparisons;
+
+        public Result(List<MetricComparison> metricComparisons)
+        {
+            requireNonNull(metricComparisons, "metricComparisons can not be null");
+            checkArgument(!metricComparisons.isEmpty(), "No metric comparisons given");
+            this.metricComparisons = ImmutableList.copyOf(metricComparisons);
+        }
+
+        public Result estimate(Metric metric, MetricComparisonStrategy strategy)
+        {
+            return testMetrics(metric, metricComparison -> metricComparison.result(strategy) == MATCH);
+        }
+
+        private Result testMetrics(Metric metric, Predicate<MetricComparison> assertCondition)
+        {
+            List<MetricComparison> testMetrics = metricComparisons.stream()
+                    .filter(metricComparison -> metricComparison.getMetric() == metric)
+                    .filter(metricComparison -> IS_OUTPUT_NODE.test(metricComparison.getPlanNode()))
```

Comment:
If we're always interested in `OutputNode`'s metrics only, we should discard anything else in the ctor.